packages/http/libraries/core/src/http/adapter/InversifyHttpAdapter.ts

```diff
@@ -188,19 +195,20 @@ export abstract class InversifyHttpAdapter<
     request: TRequest,
     response: TResponse,
     value: ControllerResponse,
+    statusCode?: HttpStatusCode,
   ): unknown {
     let body: object | string | number | boolean | undefined = undefined;
-    let statusCode: HttpStatusCode | undefined = undefined;
+    let httpStatusCode: HttpStatusCode | undefined = statusCode;
 
     if (value instanceof HttpResponse) {
       body = value.body;
-      statusCode = value.statusCode;
+      httpStatusCode = value.statusCode;
     } else {
       body = value;
     }
 
-    if (statusCode !== undefined) {
-      this._setStatus(request, response, statusCode);
+    if (httpStatusCode !== undefined) {
+      this._setStatus(request, response, httpStatusCode);
     }
 
     if (typeof body === 'string') {
```

Comment:
Maybe we should rely on the [Accept](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept) header or delegate in the integration to decide whether or not string (text/plain mime type) or json (application/json mime) are used

Reply:
That a good design question. We have 2 ways:
- Delegate to http drive
- Get agnostic response

Second option is the current implementation. idk what option is better, for me it's just a approach